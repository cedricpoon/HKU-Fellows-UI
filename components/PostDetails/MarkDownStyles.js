import { logo } from 'hkufui/theme/palette';

export default {
  /*
   * -------------------------------------------------
   * Custom styling for React Native Markdown Renderer
   * NOT following gridBase
   * -------------------------------------------------
   */
  blockquote: {
    paddingHorizontal: 10,
    marginHorizontal: 10,
    marginTop: 5,
    borderLeftWidth: 3,
    borderLeftColor: '#CCC'
  },
  link: {
    textDecorationLine: 'underline',
    color: logo.blue
  },
  text: {
    color: 'black'
  }
}
