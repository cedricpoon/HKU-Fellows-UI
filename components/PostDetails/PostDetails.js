import React, { Component } from 'react';
import { Content, Text } from 'native-base';
import PropTypes from "prop-types";

class PostSwipable extends Component {
  render() {
    const { id } = this.props.post;

    return (
      <Content>
        <Text>{id}</Text>
      </Content>
    );
  }
}

PostSwipable.propTypes = {
  post: PropTypes.shape({
    id: PropTypes.string.isRequired,
    author: PropTypes.string,
    timestamp: PropTypes.string.isRequired,
    content: PropTypes.string.isRequired
  }).isRequired
};

export default PostSwipable;
