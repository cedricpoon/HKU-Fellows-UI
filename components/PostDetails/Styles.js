import { StyleSheet } from 'react-native';

import { gridBase, fontSize, comment } from 'hkufui/theme/grid';
import { logo, seperator } from 'hkufui/theme/palette';
import customMarkDown from './MarkDownStyles';

const headlineContent = {
  marginLeft: gridBase,
  fontSize: fontSize.titleSize
};

const panel = {
  flexDirection: 'row',
  alignItems: 'center'
};

export default StyleSheet.create({
  ...customMarkDown,
  container: {
    paddingHorizontal: comment.spacingHorizontal
  },
  headline: {
    ...panel,
    justifyContent: 'space-between',
    paddingVertical: comment.spacingVertical,
    borderBottomWidth: gridBase / 10,
    borderColor: seperator,
  },
  leftPanel: {
    ...panel
  },
  rightPanel: {
    ...panel,
    alignItems: 'baseline',
    paddingLeft: gridBase,
    paddingVertical: 1,
    shadowColor: logo.black,
    shadowOffset: { width: 1, height: 1 },
    shadowOpacity: 0.1,
    shadowRadius: 1,
    borderRadius: gridBase,
    backgroundColor: 'white'
  },
  index: {
    fontSize: fontSize.miniSize,
    color: 'gray'
  },
  author: {
    ...headlineContent,
    color: logo.dimmed.green
  },
  anonymous: {
    ...headlineContent,
    color: logo.dimmed.blue,
    fontStyle: 'italic'
  },
  temperature: {
    marginRight: gridBase,
    fontSize: fontSize.semiTitleSize
  },
  hot: {
    color: logo.red
  },
  date: {
    ...headlineContent,
    fontSize: fontSize.semiTitleSize,
    color: logo.black
  },
  solved: {
    ...headlineContent,
    fontSize: fontSize.semiTitleSize,
    color: logo.dimmed.yellow
  },
  contentContainer: {
    marginVertical: gridBase
  }
});
