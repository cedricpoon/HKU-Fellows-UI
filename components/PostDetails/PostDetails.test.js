import React from 'react';
import { View } from 'react-native';
import { shallow } from 'enzyme';
import PostDetails from './PostDetails';

describe('Testing PostDetails Component', () => {
  const mockMarkdownRenderer = () => {
    return <View></View>;
  }

  const requiredProps = {
    index: 1,
    comment: {
      id: 'id',
      timestamp: 'timestamp',
      author: 'author',
      content: 'content',
      temperature: 1
    },
    markdownRenderer: mockMarkdownRenderer,
    htmlRenderer: mockMarkdownRenderer
  };

  it('renders as expected with required props', () => {
    const wrapper = shallow(
      <PostDetails {...requiredProps} />
    );
    expect(wrapper).toMatchSnapshot();
  });

  it('is selectedAnswer', () => {
    const wrapper = shallow(
      <PostDetails {...requiredProps} selectedAnswer />
    );
    expect(wrapper).toMatchSnapshot();
  });

  it('renders with markdown', () => {
    const wrapper = shallow(
      <PostDetails {...requiredProps} native />
    );
    expect(wrapper).toMatchSnapshot();
  });
});
