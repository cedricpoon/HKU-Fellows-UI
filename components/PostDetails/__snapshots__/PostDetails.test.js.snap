// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Testing PostDetails Component renders as expected with required props 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <PostSwipable
    post={
      Object {
        "author": "author",
        "content": "content",
        "id": "id",
        "timestamp": "timestamp",
      }
    }
  />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "class",
    "props": Object {
      "children": <Styled(Text)>
        id
      </Styled(Text)>,
      "virtual": undefined,
    },
    "ref": null,
    "rendered": Object {
      "instance": null,
      "key": undefined,
      "nodeType": "class",
      "props": Object {
        "children": "id",
        "virtual": undefined,
      },
      "ref": null,
      "rendered": "id",
      "type": [Function],
    },
    "type": [Function],
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "class",
      "props": Object {
        "children": <Styled(Text)>
          id
        </Styled(Text)>,
        "virtual": undefined,
      },
      "ref": null,
      "rendered": Object {
        "instance": null,
        "key": undefined,
        "nodeType": "class",
        "props": Object {
          "children": "id",
          "virtual": undefined,
        },
        "ref": null,
        "rendered": "id",
        "type": [Function],
      },
      "type": [Function],
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getDerivedStateFromProps": true,
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
  },
}
`;
